bool movestatus
int32 mileage_val
